server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java

```diff
@@ -79,41 +75,30 @@ public void init() {
     @SuppressWarnings("unchecked")
     @Override
     protected List<Domain> getDomainListInternal() throws DomainListException {
-        List<Domain> domains = new ArrayList<>();
         EntityManager entityManager = entityManagerFactory.createEntityManager();
-        final EntityTransaction transaction = entityManager.getTransaction();
         try {
-            transaction.begin();
             List<String> resultList = entityManager
                     .createNamedQuery("listDomainNames")
                     .getResultList();
-            domains = resultList
+            return resultList
                     .stream()
-                    .map(domainAsString -> Domain.of(domainAsString))
+                    .map(Domain::of)
                     .collect(Guavate.toImmutableList());
-            transaction.commit();
         } catch (PersistenceException e) {
             LOGGER.error("Failed to list domains", e);
-            rollback(transaction);
             throw new DomainListException("Unable to retrieve domains", e);
         } finally {
             entityManager.close();
         }
-        return ImmutableList.copyOf(domains);
     }
 
     @Override
     protected boolean containsDomainInternal(Domain domain) throws DomainListException {
         EntityManager entityManager = entityManagerFactory.createEntityManager();
-        final EntityTransaction transaction = entityManager.getTransaction();
         try {
-            transaction.begin();
-            boolean result = containsDomainInternal(domain, entityManager);

```

Comment:
It's probably a problem: a transaction also mean you will have a consistent view for reads during the transaction. It's the isolation feature of the MVCC model.

Reply:
I don t think so.

You have the guaranty to not see partially applied transacrions. As long as we get a single read statement, we don't need to explicitly wrap it in a transaction.

See https://stackoverflow.com/questions/26327274/do-you-need-a-database-transaction-for-reading-data

Reply:
> I don t think so.

You answer to which statement?

> You have the guaranty to not see partially applied transacrions. As long as we get a single read statement, we don't need to explicitly wrap it in a transaction.

I agree with you concerning single statements: It doesn't make a difference. However, I still claim transaction give you isolation (if you configure it that way, see https://stackoverflow.com/questions/16162357/transaction-isolation-levels-relation-with-locks-on-table). BTW, we should use at least "REPEATABLE READ".